Source/TheAviator/gherkin/ScenarioEnvironment.ts

```diff
@@ -12,47 +12,58 @@ import { Scenario } from './Scenario';
+const eventStoreDumpFolderName = 'eventStore';
+const containerOptionsFileExtension = '.json';
+
+export type MicroserviceAction = (microservice: Microservice) => Promise<void>;
 
 export class ScenarioEnvironment {
-    static empty: ScenarioEnvironment = new ScenarioEnvironment({} as IFlightPaths, new ScenarioEnvironmentDefinition(), {}, {} as ISerializer);
 
-    readonly definition: ScenarioEnvironmentDefinition;
-    readonly microservices: { [key: string]: Microservice };
+    /**
+     * Gets an empty {ScenarioEnvironment}
+     *
+     * @static
+     * @type {ScenarioEnvironment}
+     */
+    static readonly empty: ScenarioEnvironment = new ScenarioEnvironment(
+        {} as IFlightPaths,
+        new ScenarioEnvironmentDefinition(),
+        {},
+        {} as ISerializer);
 
     constructor(
-        private _flightPaths: IFlightPaths,
-        definition: ScenarioEnvironmentDefinition,
-        microservices: { [key: string]: Microservice },
+        private readonly _flightPaths: IFlightPaths,
+        readonly definition: ScenarioEnvironmentDefinition,
+        readonly microservices: { [key: string]: Microservice },
         readonly _serializer: ISerializer) {
-        this.definition = definition;
-        this.microservices = microservices;
         this.writeConfigurationFiles();
     }
 
     async start(): Promise<void> {
-        await this.forEachMicroservice(async _ => await _.start());
+        await this.forEachMicroservice(_ => _.start());
         await this.connectConsumersToProducers();
     }
 
     async stop(): Promise<void> {
         await this.disconnectConsumersFromProducers();
-        await this.forEachMicroservice(async _ => await _.kill());
+        await this.forEachMicroservice(_ => _.kill());
     }
 
-    async forEachMicroservice(method: MicroserviceMethod) {
+    forEachMicroservice(action: MicroserviceAction): Promise<void[]> {
+        const promises: Promise<void>[] = [];
         for (const microservice of Object.values(this.microservices)) {
-            await method(microservice);
+            promises.push(action(microservice));
         }
+        return Promise.all(promises);
     }
 
     async dumpEventStore(scenario: Scenario) {
         this.forEachMicroservice(async (microservice) => {
             const microserviceDestinationDirectory = this._flightPaths.forMicroserviceInScenario(scenario, microservice);
-            const destinationDirectory = path.join(microserviceDestinationDirectory, 'eventStore');
-
-            if (!fs.existsSync(destinationDirectory)) {
-                fs.mkdirSync(destinationDirectory, { recursive: true });
+            const destinationDirectory = path.join(microserviceDestinationDirectory, eventStoreDumpFolderName);
+            const directoryExists = await fs.promises.exists(destinationDirectory);
+            if (!directoryExists) {
+                await fs.promises.mkdir(destinationDirectory, { recursive: true });
             }
```

Comment:
`fs.exists()` has been [deprecated](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback) so we could just use `fs.promises.mkdir()` straight up. 

With the `recursive: true` it [won't raise an error:](https://nodejs.org/api/fs.html#fs_fs_mkdir_path_options_callback) 

>  Calling fs.mkdir() when path is a directory that exists results in an error only when recursive is false.

So we can just call it straight up or we could use `fs.stat()` or `fs.access()` on it first, but both of these methods [docs](https://nodejs.org/api/fs.html#fs_fs_stat_path_options_callback) advise against it:

> Using fs.stat() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Instead, user code should open/read/write the file directly and handle the error raised if the file is not available.

```suggestion
            await fs.promises.mkdir(destinationDirectory, { recursive: true });
```